hr_attendance_schedule/models/hr_attendance_clock.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+import pytz
+from datetime import datetime
+
+from odoo import api, fields, models
+from odoo.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class AttendanceClock(models.Model):
```

Comment:
why not inherit from AbstractModel?

Reply:
Because get_system_clock is called from the JS. Is that feasible with AbstractModel?

Reply:
absolutely. And the way I understand `_auto`, it's meant to say: "I'll take care of creating a table myself": https://github.com/odoo/odoo/blob/10.0/odoo/models.py#L2337

Reply:
I thought abstract classes could not be used unless instanciated via a derived class, or using class methods, right? Then again, that was in Java and C# but Python's concept of OOP and inheritance feels a bit loose and permissive to me. Is there an official doc about models.AbstractModel?

Reply:
models.py, linked above